### ISSUE TITLE
Request matching with an array in the body

### ISSUE THREAD

**Issue by abhayachauhan** [pact-foundation/pact-mock_service#25]
We have a request which has an array at the top level of the body which I am unable to perform a type match on.
For example:

```
        .withRequest({
            method: 'post',
            path: '/xxx/en',
            body: {
                  "json_class": "Pact::SomethingLike",
                    "contents": ["REVISION_HISTORY"]
               }
             })
```

This request only matches request with ["REVISION_HISTORY"] in the body and not all arrays.

**Comment by bethesque** [comment]
Please follow these instructions: https://github.com/realestate-com-au/pact/wiki/v2-flexible-matching

**Comment by abhayachauhan** [comment]
This being a pact spec v2 feature. 
Is PACT Mock Service PACT v2 compliant?

**Comment by bethesque** [comment]
Yes. If you are starting it from the command line, you'll need to give it the command line option, can't remember what it is off the top of my head, it'll be in the help output.

**Comment by YOU54F** [comment]
[pact mock server](https://github.com/pact-foundation/pact-core-mock-server) has been written in written in rust, avaialble as both a CLI and FFI interface for interop with client languages

If there are further issues, they should be raised in their respective repos
